### ISSUE TITLE
rend3-alacritty: DECTCEM VT510 escape code does not work

### ISSUE THREAD

**Issue by siltyy** [hearth-rs/hearth#172]
The [DECTCEM](https://vt100.net/docs/vt510-rm/DECTCEM.html) escape code doesn't work in the `rend3-alacritty` demo. DECTCEM is responsible for showing and hiding the terminal emulator's cursor and is used by pretty much every TUI application under the sun. The lack of support for it has the unfortunate side-effect of causing certain applications (I noticed it while testing [doom-ascii](https://github.com/wojciech-graj/doom-ascii)) to show the cursor jumping around onscreen as things are drawn.

This can be verified by attempting to hide the cursor with `printf '\e[?25l'` (and, on a supporting terminal, showing it again with `printf '\e[?25h'`).

It's also worth noting that alacritty itself *does* implement this properly.

**Comment by marceline-cramer** [comment]
I've ran those `printf` commands in the latest main demo and the cursor appears and disappears as expected. However, doom-ascii *sometimes* (but not always) has the flickering cursor issue.

Maybe low performance affecting a race condition is a factor here?

**Comment by lilithium-hydride** [comment]
Did a quick pull+rebuild and the behavior on my end matches yours, so I guess the main issue here is resolved now. I was on latest when i tried it before, so I guess something somewhere got updated since I filed the issue. 

I tried running doom-ascii again in foot and alacritty. foot did not exhibit the odd cursor flickering issue, but alacritty did. I also tested Konsole, and did not see any issues on there.

**Comment by siltyy** [comment]
Did a quick pull+rebuild and the behavior on my end matches yours, so I guess the main issue here is resolved now. I was on latest when i tried it before, so I guess something somewhere got updated since I filed the issue.

I tried running doom-ascii again in foot and alacritty. foot did not exhibit the odd cursor flickering issue, but alacritty did. I also tested Konsole, and did not see any issues on there.
